tests/unit/db/mysql/test_load_options.py

```diff
@@ -0,0 +1,208 @@
+                                       fields_terminated_by=",",
+                                       fields_enclosed_by='"',
+                                       fields_optionally_enclosed_by='"',
+                                       fields_escaped_by='\\',
+                                       lines_starting_by='abc',
+                                       lines_terminated_by='\r\n',
+                                       ignore_n_lines=1)
+            options.generate_load_data_sql(filename="another_filename.txt",
+                                           schema_name='myschema',
+                                           table_name='mytable')
+
+    def test_generate_load_data_sql_windows_filename(self) -> None:
+        options = MySqlLoadOptions(character_set="utf16",
+                                   fields_terminated_by=",",
+                                   fields_enclosed_by=None,
+                                   fields_optionally_enclosed_by='"',
+                                   fields_escaped_by='\\',
+                                   lines_starting_by='abc',
+                                   lines_terminated_by='\r\n',
+                                   ignore_n_lines=1)
+        sql = options.generate_load_data_sql("c:\\Some Path\\OH GOD LET IT END~1.CSV",
+                                             schema_name='myschema',
+                                             table_name='mytable')
+        # https://dev.mysql.com/doc/refman/8.0/en/load-data.html
+        #
+        # The file name must be given as a literal string. On Windows,
+        # specify backslashes in path names as forward slashes or
+        # doubled backslashes.
+        #
+        expected_sql = """\
+LOAD DATA
+LOCAL INFILE 'c:\\\\Some Path\\\\OH GOD LET IT END~1.CSV'
+INTO TABLE myschema.mytable
+CHARACTER SET 'utf16'
+FIELDS
+    TERMINATED_BY ','
+    OPTIONALLY ENCLOSED_BY '"'
+    ESCAPED BY '\\\\'
+LINES
+    STARTING BY 'abc'
+    TERMINATED BY '\\r\\n'
+IGNORE 1 LINES
+"""
+        sqltext = str(sql.compile(compile_kwargs={"literal_binds": True}))
+
+        self.assertEqual(sqltext, expected_sql)
+
+    def test_vertica_dialect_style_terminators(self) -> None:
+        options = MySqlLoadOptions(character_set="utf16",
+                                   fields_terminated_by="\002",
+                                   fields_enclosed_by=None,
+                                   fields_optionally_enclosed_by='"',
+                                   fields_escaped_by='\\',
+                                   lines_starting_by='abc',
+                                   lines_terminated_by="\001",
+                                   ignore_n_lines=1)
+        sql = options.generate_load_data_sql(filename="another_filename.txt",
+                                             schema_name='myschema',
+                                             table_name='mytable')
+        # TODO: Verify that these work
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker